services/language_translation/v2.js

```diff
@@ -104,6 +104,29 @@ module.exports = function (RED) {
     // have been provided. 
     this.on('input', function (msg) {
 
+      // The dynamic nature of this node has caused problems with the password field. it is 
+      // hidden but not a credential. If it is treated as a credential, it gets lost when there
+      // is a request to refresh the model list. 
+      //
+      // Credentials are needed for each of the modes.
+
+      username = sUsername || this.credentials.username;
+      password = sPassword || this.credentials.password || config.password;
+
+      if (!username || !password) {
+        message = 'Missing Language Translation service credentials';
+        node.error(message, msg);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["message" was used before it was defined](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2613495930/issues/source?bid=3225115&fileBranchId=3311948#l118)